kong/plugins/ldap-auth/access.lua

```diff
@@ -58,7 +67,37 @@ local function ldap_authenticate(given_username, given_password, conf)
     end
   end
 
-  is_authenticated, err = ldap.bind_request(sock, who, given_password)
+  if conf.bind_dn then
+    is_authenticated = false
+    ok, err = ldap.bind_request(sock, conf.bind_dn, conf.ldap_password)
+    if ok then
+      local search_results = ldap.search_request(sock, {
+        base = conf.base_dn;
+        scope = "sub";
+        filter = conf.attribute .. "=" .. given_username;
+        attrs = "userPassword";
+      })
+      local user, value = next(search_results)
+      if user and type(value.userPassword) == "string" then
```

Comment:
@daurnimator would following checks handle all the supported scheme like md5, sha256 etc? should we do a second bind operation with fully qualified DN, one we got from search?

Reply:
> would following checks handle all the supported scheme like md5, sha256 etc? 

They handle all the ones I saw documented by OpenLDAP.

> should we do a second bind operation with fully qualified DN, one we got from search?

No. the main issue this PR solves is that most ldap servers in the wild only have a single DN or so that can bind.

Reply:
> No. the main issue this PR solves is that most ldap servers in the wild only have a single DN or so that can bind.
 Are you sure about this? I thought general convention is let the sever handle password comparison as part of bind DN. This way you don't have to worry about scheme supported by LDAP server.  Found some internet reference https://stackoverflow.com/questions/546438/authenticating-user-using-ldap-from-php?rq=1, may not taken as best approach

Reply:
You can do a second bind, but that requires sending the user's attempted password to the server; which I see as un-optimal security.
Doing the hash+compare in the client works too :)

Reply:
Sounds good, just update the doc about limitation about supported scheme and bind_dn should have access to `userPassword` attribute. You also might want to add that `self_check` function @Tieske advised.